### PR TITLE
Handle value specified as bytes received as integer

### PR DESCRIPTION
**Pull request by marcelveldt**
Somehow some devices send some values (such as the Noc) as integer instead of bytes so our parser chokes on that.
Not sure if this is device related or somehow the tlv parser doesnt handle it but its not super big deal.
Convert it so parsing the values doesnt crash in the client.

Fixes https://github.com/home-assistant/core/issues/116304
Fixes https://github.com/home-assistant/core/issues/113279

### CODE REVIEW

File: tests/common/test_parser.py [hunk 1]
```diff
@@ -107,3 +108,6 @@ def test_dataclass_from_dict():
     # test extra keys not silently ignored in strict mode
     with pytest.raises(KeyError):
         dataclass_from_dict(BasicModel, raw2, strict=True)
+    # test int gets converted to bytes
+    res = parse_value("int_value_that_should_be_bytes", 5, bytes)
+    assert isinstance(res, bytes)
```
Comment (MartinHjelmare) [inline tests/common/test_parser.py:113]: Maybe assert what the value is instead?

**Review by MartinHjelmare — APPROVED** [review]

File: tests/common/test_parser.py (hunk 1 above)
Comment (marcelveldt) [reply tests/common/test_parser.py:113]: Yeah and I just realize this conversion is wrong
bytes(5) creates b"\x00\x00\x00\x00\x00" and not b'\x05\x00\x00\x00'

Comment (MartinHjelmare) [reply tests/common/test_parser.py:113]: https://docs.python.org/3/library/stdtypes.html#int.to_bytes

Comment (marcelveldt) [reply tests/common/test_parser.py:113]: Yeah exactly but still I wonder if we should do it this way. I just looked up the spec and this should contain the NOC as (bytes encoded) string. So a single int in this field is basically just junk and we should just ignore it. We can parse it to bytes (assuming little endian, unsigned) but its still junk.

Comment (marcelveldt) [reply tests/common/test_parser.py]: anyways, changed it so we're unblocked.

**Review by MartinHjelmare — APPROVED** [review]

File: tests/common/test_parser.py (hunk 1 above)
Comment (marcelveldt) [reply tests/common/test_parser.py]: @MartinHjelmare what do you think ? Do the conversion (as at least we get a bytes object back) or just set it to an empty bytes object ?

Comment (MartinHjelmare) [reply tests/common/test_parser.py]: What does the NOC do?

Comment (marcelveldt) [reply tests/common/test_parser.py]: > What does the NOC do?

Its a certificate

Comment (marcelveldt) [reply tests/common/test_parser.py]: So setting it to an empty string feels more safe here to me

Comment (marcelveldt) [reply tests/common/test_parser.py]: I changed it to only handle this specific edge case - it didn't feel right otherwise as we would be hiding issues.

**Review by MartinHjelmare — APPROVED** [review]
